behavex/runner.py

```diff
@@ -605,19 +604,36 @@ def launch_by_scenario(features,
                 feature_json_skeleton = scenario_information["feature_json_skeleton"]
                 scenario_line = scenario_information["scenario_line"]
                 scenario_outline_line = scenario_information["scenario_outline_line"]
-                future = process_pool.submit(execute_tests,
-                                             features_path=features_path,
-                                             feature_filename=feature_filename,
-                                             feature_json_skeleton=feature_json_skeleton,
-                                             scenarios_to_run_in_feature=scenarios_to_run_in_feature,
-                                             scenario_outline_line=scenario_outline_line,
-                                             scenario_lines=scenario_line,
-                                             multiprocess=True,
-                                             config=ConfigRun(),
-                                             lock=lock,
-                                             shared_removed_scenarios=shared_removed_scenarios
-                                             )
-                parallel_processes.append(future)
+                
+                if is_parallel_on_scenario:
+                    future = process_pool.submit(execute_tests,
+                                features_path=features_path,
+                                feature_filename=feature_filename,
+                                feature_json_skeleton=feature_json_skeleton,
+                                scenarios_to_run_in_feature=scenarios_to_run_in_feature,
+                                scenario_outline_line=scenario_outline_line,
+                                scenario_lines=scenario_line,
+                                multiprocess=True,
+                                config=ConfigRun(),
+                                lock=lock,
+                                shared_removed_scenarios=shared_removed_scenarios
+                                )
+                    parallel_processes.append(future)
+                else:
+                    for line in scenario_line:
+                        future = process_pool.submit(execute_tests,
+                                features_path=features_path,
+                                feature_filename=feature_filename,
+                                feature_json_skeleton=feature_json_skeleton,
+                                scenarios_to_run_in_feature=scenarios_to_run_in_feature,
+                                scenario_outline_line=scenario_outline_line,
+                                scenario_lines=[line],
```

Comment:
@hrcorval , i ended up not using `scenario_outline_line` - im not sure what that was intended for in your POC.

but just looping over the scenario lines works. (see the loop on line `623`) then i just pass a single item list as `scenario_lines=[line]`

i can see behave runs the tests properly this way but i can't figure out why the `execution_codes` and `json_reports` generated is incomplete when i do this approach.

## **given** a feature file:
```feature
Feature: Example Table Tags

  Background:
    Given I have something

  Scenario Outline: Foo From Example 1
    Then I expect something

    Examples:
      | data |
      | 1    |
      | 2    |

  Scenario Outline: Foo From Example 2
    Then I expect something

    Examples:
      | data |
      | 1    |
      | 2    |
```

## **when** i run `parallel-scheme=examples`
## **then** i expect this on 
![image](https://github.com/user-attachments/assets/b4aac8fd-f9bb-4165-b681-85cdfea6804b)
![image](https://github.com/user-attachments/assets/6c30a0d4-a77a-409c-8fd3-987ad171a292)

## Actual
![image](https://github.com/user-attachments/assets/3cc9a153-715d-43e7-b0e6-562a2646385e)

This only contains information on the last example on the example table.

Reply:
would appreciate if you can provide some guidance here. thank you!